integration/actions/test_test.go

```diff
@@ -0,0 +1,118 @@
+//go:build actions
+// +build actions
+
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actions
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/okteto/okteto/integration"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	testPath = "okteto/test"
+)
+
+func TestOktetoTestActions(t *testing.T) {
+	integration.SkipIfWindows(t)
+	namespace := integration.GetTestNamespace(t.Name())
+
+	assert.NoError(t, executeCreateNamespaceAction(namespace))
+
+	oktetoManifest, err := createOktetoTestManifest(t)
+	assert.NoError(t, err)
+
+	assert.NoError(t, executeOktetoTestAction(namespace, oktetoManifest))
+}
+
+func executeOktetoTestAction(namespace, oktetoManifest string) error {
+	actionRepo := fmt.Sprintf("%s%s.git", githubHTTPSURL, testPath)
+	actionFolder := strings.Split(testPath, "/")[1]
+	log.Printf("cloning destroy path repository: %s", actionRepo)
+	if err := integration.CloneGitRepoWithBranch(actionRepo, oktetoVersion); err != nil {
+		if err := integration.CloneGitRepo(actionRepo); err != nil {
+			return err
+		}
+		log.Printf("cloned repo %s main branch\n", actionRepo)
+	}
+	log.Printf("cloned repo %s \n", actionRepo)
+	defer integration.DeleteGitRepo(actionFolder)
+
+	builder := exec.Command("go", "build", "-o", "test", "./cmd/main.go")
```

Comment:
I remember you mentioned something about you prefer to do this instead of just a `run main.go`, what was it?

Reply:
The dockerfile builds the action and after that executes the binary. I wanted to have it as similar as I can so if we break that step, we can have a test detecting it although we already have an specific test for it in the action repo